packages/client/test/cli/cli.spec.ts

```diff
@@ -97,17 +401,281 @@ describe('[CLI]', () => {
+    return new Promise((resolve) => {
+      child.stdout.on('data', async (data) => {
+        const message: string = data.toString()
+        if (message.includes('Client started successfully')) {
+          assert.ok(
+            message.includes('Client started successfully'),
+            'Clients started with experimental feature options'
+          )
+          child.kill()
+          resolve(undefined)
+        }
+        if (message.toLowerCase().includes('error')) {
+          child.kill(9)
+          assert.fail(`client encountered error: ${message}`)
+        }
+      })
+      child.stderr.on('data', (data) => {
+        const message: string = data.toString()
+        assert.fail(`stderr: ${message}`)
+      })
+      child.on('close', (code) => {
+        if (typeof code === 'number' && code > 0) {
+          assert.fail(`child process exited with code ${code}`)
+        }
+      })
+    })
+  }, 18000)
+  // client execution limits tests
+  it('should start client when passed options for client execution limits', async () => {
+    const file = require.resolve('../../dist/bin/cli.js')
+    const cliArgs = [
+      '--port=30304',
+      '--numBlocksPerIteration=2',
+      '--txLookupLimit=2',
+      '--maxPerRequest=2',
+      '--maxFetcherJobs=2',
+      '--minPeers=2',
+      '--maxPeers=2',
+      '--startBlock=0',
+      '--dev=poa',
+    ]
+    const child = spawn(process.execPath, [file, ...cliArgs])
+    return new Promise((resolve) => {
+      child.stdout.on('data', async (data) => {
+        const message: string = data.toString()
+        if (message.includes('Client started successfully')) {
+          assert.ok(
+            message.includes('Client started successfully'),
+            'Clients starts with client execution limits options'
+          )
+          child.kill()
+          resolve(undefined)
+        }
+        if (message.toLowerCase().includes('error')) {
+          child.kill(9)
+          assert.fail(`client encountered error: ${message}`)
+        }
+      })
+      child.stderr.on('data', (data) => {
+        const message: string = data.toString()
+        assert.fail(`stderr: ${message}`)
+      })
+      child.on('close', (code) => {
+        if (typeof code === 'number' && code > 0) {
+          assert.fail(`child process exited with code ${code}`)
+        }
+      })
+    })
+  }, 18000)
+  // Network protocol tests
+  it('should start client with custom network parameters', async () => {
+    const file = require.resolve('../../dist/bin/cli.js')
+    const cliArgs = [
+      '--rpc',
+      '--port=2100',
+      '--extIP=0.0.0.0',
+      '--rpcCors=https://foo.example',
+      '--dnsAddr=8.8.8.8',
+      '--dev=poa',
+    ]
+    const child = spawn(process.execPath, [file, ...cliArgs])
+    return new Promise((resolve) => {
+      child.stdout.on('data', async (data) => {
+        const message: string = data.toString()
+        if (message.includes('Client started successfully')) {
```

Comment:
Can you validate the `Server listener up` instead and verify that we're getting the correct `extIP` and `port`?
The log message will look like:
`[07-27|10:02:22] INFO Server listener up transport=rlpx url=enode://7a03e93feceb3f9905f82c2fe498eb4d13c48ad62e3880b79a7cc8c0021e32487b11e5318044e9a5617d9e977dba9633bcf63bce60dbf5aff0b4caee4078a798@0.0.0.0:2100`

YOu can just split the string on the `@` and parse the IP and port address to verify.

Reply:
fixed